### PR TITLE
stdlib/3/ast: add visit_* methods to NodeVisitor

### PR DESCRIPTION
**Pull request by bluetech**
NodeVisitor recurses over an AST tree. When encountering a node, it
checks if a method called `visit_{node.__class__.__name__}` exists, and
calls it if so, otherwise calls the generic visitor.

Add the possible methods to NodeVisitor. This is not exactly correct,
since the methods don't *actually* exist on NodeVisitor, e.g.
`NodeVisitor().visit_Module(...)` doesn't work. But it's nice for
subclasses to know which methods they can override and which type they
should have.

### CODE REVIEW

**Comment by bluetech** [comment]
This is re. #3767 (cc @hauntsaninja). I think it's either this (have all) or remove the `visit_Constant` (have none).

**Comment by bluetech** [comment]
Hmm stubtest complains "`ast.NodeVisitor.visit_withitem` is not present at runtime" (etc.), which is true, as explained. I suppose once backward compat for `Constant` is removed, it will also fail for `visit_Constant`.

**Comment by srittau** [comment]
I think it makes sense to have these defined for the reason you stated. We probably need stubtest overrides.

File: stdlib/3/ast.pyi [hunk 1]
```diff
@@ -19,11 +19,128 @@ else:
 class NodeVisitor:
     def visit(self, node: AST) -> Any: ...
     def generic_visit(self, node: AST) -> Any: ...
-    if sys.version_info >= (3, 8):
+    def visit_Module(self, node: Module) -> Any: ...
+    def visit_Interactive(self, node: Interactive) -> Any: ...
+    def visit_Expression(self, node: Expression) -> Any: ...
+    def visit_Suite(self, node: Suite) -> Any: ...
```
Comment (isidentical) [inline stdlib/3/ast.pyi]: There is an open issue about this in the bugtracker, https://bugs.python.org/issue39639

Comment (bluetech) [reply stdlib/3/ast.pyi]: I did wonder what this one was. If it's removed from python, will need to be made conditional I suppose.

**Comment by bluetech** [comment]
I added the methods to the stubtest whitelist, and it passes now.

**Pull request merged**